### PR TITLE
Residence integration + configurable "No ... near shop" messages

### PR DESCRIPTION
**Pull request by SengyEU**

### CODE REVIEW

**Comment by myzticbean** [comment]
@SengyEU I am good with the rest of the changes. Please address the above mentioned comments, and its good to merge.

**Comment by SengyEU** [comment]
Wdym? I dont see any comments anywhere...

**Comment by myzticbean** [comment]
> Wdym? I dont see any comments anywhere...

@SengyEU I tagged ur name now in the review comments, see if you are able to view now.

**Comment by SengyEU** [comment]
Nope, i dont think the comments show up until you submit the review

**Review by myzticbean — CHANGES_REQUESTED** [review]

**Comment by myzticbean** [comment]
> Nope, i dont think the comments show up until you submit the review

sorry my bad, just did

File: pom.xml [hunk 1]
```diff
@@ -372,5 +372,12 @@
             <version>0.4.7</version>
             <scope>provided</scope>
         </dependency>
+        <dependency>
+            <groupId>zrips.residence</groupId>
+            <artifactId>Residence</artifactId>
+            <version>5.1.5.1</version>
+            <scope>system</scope>
+            <systemPath>${project.basedir}/lib/Residence5.1.5.1.jar</systemPath>
```
[the remark this answers is not in the document]
Comment (SengyEU) [reply pom.xml:380]: oh yea, the gitignore must have taken it out

File: README.md [hunk 2]
```diff
@@ -49,6 +49,7 @@ command-alias: []
   - If you added a new warp and want it to get updated immediately, run **/finditemadmin reload**
   - Remember, this applies only to Essential Warps.
 - WorldGuard region support for fetching the WorldGuard region the shop is in (if overlapping regions, highest priority will be chosen)
+- Residence support for fetching the residence the shop is in (including subzones)
```
[the remark this answers is not in the document]
Comment (SengyEU) [reply README.md:52]: 👍

**Comment by SengyEU** [comment]
Here you go

**Review by myzticbean — APPROVED** [review]

**Pull request merged**